### PR TITLE
[RFC] Advertise the Jabber room instead of the IRC channel

### PR DESCRIPTION
**Pull request by linkmauve**
There are a few things I’d like to fix before getting this patch merged:
- Host the web client on a tatoeba.org domain, so it doesn’t go through my domain.
- Add a qrcode to the room, for mobile clients (at least [Conversations](http://conversations.im/) supports that).
- Add a **small** list of suggested clients.
- Make the web client only appear for logged users, and default their nick to their Tatoeba one.

### CODE REVIEW

File: app/views/pages/contact.ctp [hunk 1]
```diff
@@ -93,12 +93,16 @@ $this->set('title_for_layout', $pages->formatTitle(__('Contact us', true)));
...
         <?php
         echo format(
             __(
-                'We also have an IRC channel on freenode, #tatoeba. If you are not '.
-                'familiar with IRC, you can read the <a href="{}">Help</a>.', true
+                'We also have a Jabber room, <a '.
+                'href="xmpp:tatoeba@chat.tatoeba.org?join">'.
+                'tatoeba@chat.tatoeba.org</a>, also available with the web'.
+                'client below.  If you are not familiar with Jabber, you '.
+                'can read the <a href="{}">Help</a>.', true
```
Comment (jiru) [inline app/views/pages/contact.ctp]: You should put the href of the first link out of the string so that it’s easier to translate. See [this example](https://github.com/Tatoeba/tatoeba2/blob/39fbbb5301ce10423dbd591d5c08e9382ce859d0/app/views/pages/donate.ctp#L36-L44).

**Comment by halfdan** [comment]
@linkmauve Correct me if I'm wrong, but it seems that IRC has long been abandoned and Jabber was replaced by Gitter. While I'm a strong advocate of XMPP/Jabber I don't think we'd get anyone on there and Gitter is now the preferred solution. We can probably close this PR?

**Comment by halfdan** [comment]
This said - we should totally replace the IRC message on tatoeba.org with a reference to the Gitter room.

**Comment by Ppjet6** [comment]
The IRC channel was never abandonned, and both the IRC side and the XMPP side have always been bridged, (meaning you can join one or the other indifferently).

From what I have seen of the gitter room, there's been not much activity at all since the beginning.

The move to gitter was probably to try and grasp the attention of developers willing to contribute, (which I totally respect), and also probably a social issue, because there was a lot of off-topic on IRC. Though, having been one of those on the XMPP side of the room for a long time, I feel the latter is a non-issue, and I welcome people who have tatoeba at heart in the chatroom.

👍 for making tatoeba support easily accessible.

**Comment by trang** [comment]
> The move to gitter was probably to try and grasp the attention of developers willing to contribute

Actually it was not targeted to developers only. 

The main reason for creating a Tatoeba chatroom on Gitter was to have a space for real-time conversations that is not as volatile as IRC, where people can read previous conversations. To avoid for instance cases where people are logging in, seeing nothing happening and just logging out.

Also, for many people, IRC is kinda scary and complicated, not very user friendly. So it was an attempt to find something a little bit more welcoming for the non-tech members.

**Comment by linkmauve** [comment]
> The main reason for creating a Tatoeba chatroom on Gitter was to have a space for real-time conversations that is not as volatile as IRC, where people can read previous conversations.

This has always been an advertised feature of the XMPP side of the room.

Please test at https://candy.linkmauve.fr/tatoeba@chat.tatoeba.org (and then we can talk about deploying Converse directly on Tatoeba).

File: app/View/Pages/contact.ctp [hunk 2]
```diff
@@ -93,20 +93,34 @@ $this->set('title_for_layout', $this->Pages->formatTitle(__('Contact us')));
...
+                'tatoeba@chat.tatoeba.org</a>, also available with the web '.
+                'client below.  If you are not familiar with XMPP, you '.
+                'can read the <a href="{help}">Help</a>.', true
             ),
-            $this->Html->url(
-                array(
-                    "controller" => "pages",
-                    "action" => "help"
+            array(
+                'room' => "xmpp:tatoeba@chat.tatoeba.org?join"
```
Comment (trang) [inline app/View/Pages/contact.ctp]: There's a comma missing at the end. But don't push any fix yet. We're gonna need to discuss about what to write on this Contact page :)

**Comment by trang** [comment]
A few things to consider here:

1. I'm in favor of ditching IRC. There's way too much spam these days and we weren't very active anyway. We should still make sure everyone who somehow still hangs around in IRC is aware that we're moving away.
2. I'm not entirely sure if we want to embed the chat into the contact page. It's not looking too bad, but it feels a bit intrusive somehow. Maybe that's just me.

Screenshot of what it looks like currently:

contact-page

**Contact page**

I would reword the "Join us on XMPP" to something like this:

> ### Join our chatroom
>
> We have an **XMPP room**: [tatoeba@chat.tatoeba.org](link-to-xmpp). You can join with your favorite XMPP client or [from your browser](https://chat.tatoeba.org).

**Help page**

* In the first block in the main content area, I would change "... or join us on XMPP" to "... or join our [chatroom](https://chat.tatoeba.org)" (with a link to the chatroom, just like there's a link to the Wall).
* I would remove the whole "How to use XMPP" section. It feels unnecessary.
* I would also remove the whole "Contact us" section. We have a Contact page for that.
* In the "Need more help?" section, I would add a link to the contact page "...do no hesitate to [contact us](https://tatoeba.org/eng/contact)".

I'll ask @alanfgh to validate this text :)

**Comment by alanfgh** [comment]
Your suggestions sound fine to me, including the wording. There's just a typo in your last bullet point: you wrote "do no hesitate" where I'm sure you meant "do not hesitate" (which is what the text currently says).

Note aside from being intrusive, embedding the chat into the page means that whatever is being posted in the chatroom will seem to be a part of the Tatoeba website. That could easily include spam, rude language, etc.

**Comment by jiru** [comment]
> I'm in favor of ditching IRC. There's way too much spam these days and we weren't very active anyway. We should still make sure everyone who somehow still hangs around in IRC is aware that we're moving away.

Just ask @liori to change the topic of the IRC channel to something that includes a link to the new chatroom.

> I'm not entirely sure if we want to embed the chat into the contact page. It's not looking too bad, but it feels a bit intrusive somehow. Maybe that's just me.

I have the same feeling. How about just having a text that links to https://chat.tatoeba.org/? It would also feel more consistent with the other sections of the page (email and wall).

File: README.md [hunk 3]
```diff
@@ -42,12 +42,12 @@ Need help?
...
+message in our [Google group][5], ask in our [XMPP chatroom][6],
+[tatoeba@chat.tatoeba.org](xmpp:tatoeba@chat.tatoeba.org?join).
 
 [1]: https://github.com/Tatoeba/tatoeba2/wiki/Joining-the-dev-team
 [2]: https://github.com/Tatoeba/imouto
 [3]: https://github.com/Tatoeba/imouto/blob/master/README.md#imouto-for-development
 [4]: https://github.com/Tatoeba/tatoeba2/wiki/How-to-install-Tatoeba
 [5]: http://groups.google.com/group/tatoebaproject
-[6]: https://gitter.im/Tatoeba/tatoeba2?utm_source=share-link&utm_medium=link&utm_campaign=share-link
+[6]: https://chat.tatoeba.org/
```
Comment (trang) [inline README.md]: I just noticed you also edited the README. I wouldn't remove the link to Gitter just yet but simply replace the IRC part by XMPP.

In other words:
1. Keep the Gitter badge at the top. Feel free to add another badge for XMPP though.
2. Change the **Need help?** text as followed: "... don't hesitate to post a message in our [Google group][1], ask in our [Gitter chatroom][2], or in our [XMPP chatroom][3]."

[1]: http://groups.google.com/group/tatoebaproject
[2]: https://gitter.im/Tatoeba/tatoeba2?utm_source=share-link&utm_medium=link&utm_campaign=share-link
[3]: https://chat.tatoeba.org/

File: app/View/Pages/help.ctp [hunk 4]
```diff
@@ -40,55 +40,15 @@ $this->set('title_for_layout', $this->Pages->formatTitle(__('Help')));
...
-        echo __(
-            'For those who are not familiar with IRC, it is pretty much like an '.
-            'instant messenger. You will be able to talk to us in real time, '.
-            'if we are there.'
+        echo format(
+            __(
+                'If you cannot find the answer to your question, do not hesitate '.
+                'to <a href="{}">contact us</a>.'
+            ),
+            $this->Html->url(array('controller' => 'contact'))
```
Comment (trang) [inline app/View/Pages/help.ctp]: This will give a wrong URL. It should be:
```
$this->Html->url(array('controller' => 'pages', 'action' => 'contact'))
```

File: app/View/Pages/help.ctp [hunk 5]
```diff
@@ -107,10 +67,16 @@ $this->set('title_for_layout', $this->Pages->formatTitle(__('Help')));
                 __(
                     'If you have no idea what to do now that you are registered, '.
                     'you can introduce yourself on the <a href="{}">Wall</a>, or '.
```
Comment (trang) [inline app/View/Pages/help.ctp]: You should change `<a href="{}">Wall</a>` to `<a href="{wall}">Wall</a>` in order to refer to it below.

File: app/View/Pages/help.ctp [hunk 6]
```diff
@@ -107,10 +67,16 @@ $this->set('title_for_layout', $this->Pages->formatTitle(__('Help')));
                 __(
                     'If you have no idea what to do now that you are registered, '.
                     'you can introduce yourself on the <a href="{}">Wall</a>, or '.
-                    'join us on <strong>IRC</strong>. We will give you a purpose. :)',
+                    'join our <a href="{chatroom}">chatroom</strong>. We will '.
```
Comment (trang) [inline app/View/Pages/help.ctp]: The closing tag should be `</a>`, not `</strong>`.

File: app/View/Pages/help.ctp [hunk 7]
```diff
@@ -107,10 +67,16 @@ $this->set('title_for_layout', $this->Pages->formatTitle(__('Help')));
...
+                    'give you a purpose. :)',
                     true
                 ),
-                $this->Html->url(array('controller' => 'wall'))
+                $this->Html->url(
+                    array(
+                        'controller' => 'wall',
+                        'chatroom' => 'https://chat.tatoeba.org'
+                    )
+                )
```
Comment (trang) [inline app/View/Pages/help.ctp]: You need to create an array, since there are 2 variables in the string.
```
array(
    'wall' => $this->Html->url(array('controller' => 'wall')),
    'chatroom' => 'https://chat.tatoeba.org'
)
```

**Comment by trang** [comment]
Thanks @linkmauve, I added a few last comments. I can merge once those are fixed :)

> There are localisation issues though, how can https://chat.tatoeba.org/ obtain the language of the user to enable the translation in this locale?

With `$this->params['lang']`.

**Comment by linkmauve** [comment]
Everything should now be fixed.
